src/PolykeyClient.ts

```diff
@@ -35,15 +37,19 @@ interface PolykeyClient extends CreateDestroyStartStop {}
 class PolykeyClient {
   static async createPolykeyClient({
     nodePath = config.defaultsUser.nodePath,
-    streamFactory,
+    wsPort,
+    wsHost,
+    wsConfig,
     streamKeepAliveTimeoutTime,
     parserBufferByteLimit,
     fs = require('fs'),
     logger = new Logger(this.name),
     fresh = false,
   }: {
     nodePath?: string;
-    streamFactory: StreamFactory;
+    wsPort: number;
+    wsHost: string;
+    wsConfig?: Partial<WebSocketConfig>;
     streamKeepAliveTimeoutTime?: number;
     parserBufferByteLimit?: number;
```

Comment:
Change the signature so:

1. All required parameters up top
2. All optional parameters as `options = {}`
3. See `PolykeyAgent` - `options?: DeepPartial<PolykeyAgentOptions>;`

Reply:
the `wsConfig` is wholesale optional here. We shouldn't do this. `Polykeyclient` should be providing the custom verification parameters. It should be up to the `PolykeyClient` to set up all of the verification logic.

Reply:
Any options can be spread into a general `options?: DeepPartial<PolykeyClientOptions>`. We're lifting the abstraction to the PK Client.

Reply:
@addievo declare a type above `PolykeyClientOptions` with the relevant options of PolyKey client, again, refer to `PolykeyAgent.ts`.

Reply:
Done

Reply:
It is not done, you must provide a spec first and then we can review it.